File: app/src/main/java/com/booboot/vndbandroid/dao/VNWithRelations.java
package com.booboot.vndbandroid.dao;

import android.arch.persistence.room.Embedded;
import android.arch.persistence.room.Relation;
import com.booboot.vndbandroid.model.vndb.Screen;
import com.booboot.vndbandroid.model.vndb.VN;

import java.util.List;

public class VNWithRelations {
    @Embedded
    public VN vn;

    @Relation(parentColumn = "id", entityColumn = "vnId", entity = Screen.class)
    public List<Screen> screens;
}

File: app/src/main/java/com/booboot/vndbandroid/dao/VNWithRelations.kt
package com.booboot.vndbandroid.dao

import android.arch.persistence.room.Embedded
import android.arch.persistence.room.Relation
import com.booboot.vndbandroid.model.vndb.Screen
import com.booboot.vndbandroid.model.vndb.VN

class VNWithRelations {
    @Embedded
    var vn: VN? = null

    @Relation(parentColumn = "id", entityColumn = "vnId", entity = Screen::class)
    var screens: List<Screen>? = null
}